keyboards/dm9records/plaid/keymaps/stephen-huan/keymap.c

```diff
@@ -0,0 +1,372 @@
+          return;
+        }
+        break;
+      case LEDMODE_ENTER:
+        if (keycode==KC_ENT) {
+          writePinHigh(led);
+        }
+        else {
+          writePinLow(led);
+        }
+        break;
+
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  /* If the either led mode is keypressed based, call the led updater
+     then let it fall through the keypress handlers. Just to keep 
+     the logic out of this procedure */
+  if (led_config.red_mode >= LEDMODE_MODS && led_config.red_mode <= LEDMODE_ENTER) {
+      led_keypress_update(LED_RED, led_config.red_mode, keycode, record);
+  }
+  if (led_config.green_mode >= LEDMODE_MODS && led_config.green_mode <= LEDMODE_ENTER) {
+      led_keypress_update(LED_GREEN, led_config.green_mode, keycode, record);
+  }
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+    case LED_1:
+      if (record->event.pressed) {
+        if (led_config.red_mode==LEDMODE_ON) {
+            led_config.red_mode=LEDMODE_OFF;
+            writePinLow(LED_RED);
+        }
+        else {
+            led_config.red_mode=LEDMODE_ON;
+            writePinHigh(LED_RED);
+        }
+      }
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_2:
+      if (record->event.pressed) {
+        if (led_config.green_mode==LEDMODE_ON) {
+            led_config.green_mode=LEDMODE_OFF;
+            writePinLow(LED_GREEN);
+        }
+        else {
+            led_config.green_mode=LEDMODE_ON;
+            writePinHigh(LED_GREEN);
+        }
+      }
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_3:
+      led_config.red_mode=LEDMODE_MODS;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_4:
+      led_config.green_mode=LEDMODE_MODS;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_5:
+      led_config.red_mode=LEDMODE_BLINKIN;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_6:
+      led_config.green_mode=LEDMODE_BLINKIN;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_7:
+      led_config.red_mode=LEDMODE_KEY;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_8:
+      led_config.green_mode=LEDMODE_KEY;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_9:
+      led_config.red_mode=LEDMODE_ENTER;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+    case LED_0:
+      led_config.green_mode=LEDMODE_ENTER;
+      eeconfig_update_user(led_config.raw);
+      return false;
+      break;
+  }
+  return true;
```

Comment:
If you remove the "return false" from all of the macros above, you could run this: 
```c
    if (keycode >= LED_0 && keyboard <= LED_9) {
        eeconfig_update_user(led_config.raw);
    }
```

Rather than having it repeated a bunch of times. 


Also, no need to return false here. You're not replacing any functionality, and this may actually prevent cleanup from occurring (eg, one shot keys).